django/db/models/query.py

```diff
@@ -1040,6 +1041,16 @@ def _extract_model_params(self, defaults, **kwargs):
             )
         return params
 
+    def _possibly_inject_fk_val(self, kwargs):
+        """
+        Inject the foreign key field value into kwargs if the field is known
+        from a related manager context via _known_related_objects.
+        """
+        if self._known_related_objects:
+            field, related_objects = next(iter(self._known_related_objects.items()))
+            rel_obj = next(iter(related_objects.values()))
+            kwargs[field.name] = rel_obj
```

Comment:
Is there a reason we are picking the first field and the first instance for this field? It will work in the test case but `_known_related_objects` has a type of `{field: {pk: Model}}` which can contain multiple fields and multiple instances when Queryset combination is used (see usage of `_merge_known_related_objects`)

https://github.com/django/django/blob/bc1bfe12b613334bd625aeb36fd44af96d186c10/django/db/models/query.py#L2008-L2013

In other words, I don't think we can assume that `_known_related_objects` will only contain a single field and related object as assigned by `create_reverse_many_to_one_manager`.

Reply:
So from the queryset we lose reference to `self.field` and `self.instance` on the `RelatedManager`. These are used by that manager to automatically pass in the related instance to `.create()`, `.get_or_create()`,  and `.update_or_create()`.

I'd merely seen that we already have access to these on the queryset via `_known_related_objects` and thought that could be a useful starting point for investigation. Also I'd seen:

https://github.com/django/django/blob/8b241f84e25f679c459393dab2947c8354eb01a9/django/db/models/fields/related_descriptors.py#L730-L732

I guess that we'll just have to duplicate this stuff and explicitly set another attribute on `QuerySet` that we can use? Do you have any other ideas? 🤔